extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/FastBootHibernatePersistenceProvider.java

```diff
@@ -323,6 +342,12 @@ private RuntimeSettings buildRuntimeSettings(String persistenceUnitName, Recorde
         return runtimeSettingsBuilder.build();
     }
 
+    private static boolean isPostgresOrDB2(BuildTimeSettings buildTimeSettings) {
+        Object dbKind = buildTimeSettings.get(JAKARTA_HBM2DDL_DB_NAME);
```

Comment:
This may not be set, in particular when using multitenancy...

You already derive the `SupportedDatabaseKind` from the (complex) information available at build time, so maybe you could just store that in the PUDescriptor, next to the (string) `dbKind`?

Reply:
See [this commit](https://github.com/quarkusio/quarkus/pull/48130/commits/c358496337c4c28faa49f818adbec7f13004eb13) see if it's correct